app/controllers/katello/api/v2/host_tracer_controller.rb

```diff
@@ -2,50 +2,28 @@ module Katello
   class Api::V2::HostTracerController < Api::V2::ApiController
     before_action :find_host, :only => :index
 
+    resource_description do
+      api_version 'v2'
+      api_base_url "/api"
+    end
+
     api :GET, "/hosts/:host_id/traces", N_("List services that need restarting on the host")
     param :host_id, :number, :required => true, :desc => N_("ID of the host")
     def index
       collection = scoped_search(index_relation, :application, :asc, :resource_class => ::Katello::HostTracer)
       respond_for_index(:collection => collection)
     end
 
-    api :PUT, "/traces/resolve", N_("Resolve Traces")
+    api :PUT, "/hosts/:host_id/traces/resolve", N_("Resolve Traces")
+    param :host_id, :number, :required => true, :desc => N_("ID of the host")
     param :trace_ids, Array, :required => true, :desc => N_("Array of Trace IDs")
     def resolve
       traces = Katello::HostTracer.resolvable.where(id: params[:trace_ids])
```

Comment:
This code currently allows resolving traces for hosts other than --host-id !

The code should be something like: `@host.traces.resolvable.where(id: params[:trace_ids])`

And if `traces` ends up being empty, I think the API should throw an error

Reply:
Call `.to_sql` on the above example to see how it affects the query